src/components/CandidateContest.tsx

```diff
@@ -79,194 +86,162 @@ interface Props {
   updateVote: UpdateVoteFunction
 }
 
-interface State {
-  attemptedOvervoteCandidate: OptionalCandidate
-  candidatePendingRemoval: OptionalCandidate
-  isScrollAtBottom: boolean
-  isScrollAtTop: boolean
-  isScrollable: boolean
-  writeInCandateModalIsOpen: boolean
-  writeInCandidateName: string
-}
-
-const initialState = {
-  attemptedOvervoteCandidate: undefined,
-  candidatePendingRemoval: undefined,
-  isScrollable: false,
-  isScrollAtBottom: true,
-  isScrollAtTop: true,
-  writeInCandateModalIsOpen: false,
-  writeInCandidateName: '',
-}
-
-class CandidateContest extends React.Component<Props, State> {
-  public context!: React.ContextType<typeof BallotContext>
-  public state: State = initialState
-  private scrollContainer = React.createRef<HTMLDivElement>()
-
-  public componentDidMount() {
-    this.updateContestChoicesScrollStates()
-    window.addEventListener('resize', this.updateContestChoicesScrollStates)
-  }
-
-  public componentDidUpdate(prevProps: Props) {
-    /* istanbul ignore else */
-    if (this.props.vote.length !== prevProps.vote.length) {
-      this.updateContestChoicesScrollStates()
+const findCandidateById = (candidates: Candidate[], id: string) =>
+  candidates.find((c) => c.id === id)
+
+const normalizeCandidateName = (name: string) =>
+  name.trim().replace(/\t+/g, ' ').replace(/\s+/g, ' ')
+
+const CandidateContest = ({ contest, parties, vote, updateVote }: Props) => {
+  const context = useContext(BallotContext)
+  const scrollContainer = useRef<HTMLDivElement>(null) // eslint-disable-line no-restricted-syntax
+
+  const [attemptedOvervoteCandidate, setAttemptedOvervoteCandidate] = useState<
+    OptionalCandidate
+  >(undefined)
+  const [candidatePendingRemoval, setCandidatePendingRemoval] = useState<
+    OptionalCandidate
+  >(undefined)
```

Comment:
As in `YesNoContest`, you can omit the default value:

```suggestion
  const [attemptedOvervoteCandidate, setAttemptedOvervoteCandidate] = useState<
    Candidate
  >()
  const [candidatePendingRemoval, setCandidatePendingRemoval] = useState<
    Candidate
  >()
```

Reply:
thx. done.